contributing/ansible-development.rst

```diff
@@ -0,0 +1,98 @@
+deployment. A typical role structure is shown below::
+
+    defaults/           # Default variables
+    handlers/           # Handlers
+    meta/               # Role metadaa
+        main.yml        # Dependencies and Galaxy metadata
+    molecule/           # Test
+    tasks/              # Main list of tasks to be executed
+        main.yml
+    templates/          # Role templates
+    .travis.yml         # CI/deployment configuration file
+    README.md
+
+Versioning
+----------
+
+.. _PEP440: https://www.python.org/dev/peps/pep-0440/#semantic-versioning
+
+Ansible roles must follow the PEP440_ scheme for versioning. Final releases
+must also be compliant with `Semantic Versioning`_ i.e. a final version must
+be expressed as `MAJOR.MINOR.PATCH` where:
+
+- the MAJOR version must be incremented when incompatible API changes are made,
+- the MINOR version must be incremented when functionality is added in a
+  backwards-compatible manner, and
+- the PATCH version must be incremented when you backwards-compatible bug
+  fixes are made.
+
+Final releases must be tagged with a tag matching the version i.e. 
+`MAJOR.MINOR.PATCH` with no prefix for integration with Galaxy.
+
+Development releases or pre-releases must follow the PEP440_ scheme e.g.
+`MAJOR.MINOR.PATCH.dev1`, `MAJOR.MINOR.PATCHa1` `MAJOR.MINOR.PATHrc1`.
+
+Testing and Continuous Integration
+----------------------------------
+
+.. _Molecule: https://molecule.readthedocs.io/
+
+For each Ansible role, a :file:`molecule` folder should be configured allowing
+the testing to be tested using  Molecule_. One or multiple scenarios should be
+configured using at least a Docker driver if possible. A generic
+:file:`molecule` folder can be initialized using the following command::
+
+    molecule init scenario -r ansible-role-<NAME> -s default -d docker
+
+
+Continuous Integration of Ansible roles is performed using Travis CI.
+
+.. note::
+   OME Ansible roles are getting progressively upgraded from Molecule 1.x to 
+   Molecule 2.x. New roles should be configured using Molecule 2.x.
+
+Distribution and support
+------------------------
+
+All core OME Ansible roles should be deployed to
+`Ansible Galaxy <https://galaxy.ansible.com>`_ under the
+`openmicroscopy <https://galaxy.ansible.com/openmicroscopy/>`__ organization.
+All roles must support EL 7 as a primary platform. New roles should also
+include Ubuntu as a supported platform whenever possible.
+
+Ansible playbooks can consume these roles using a :file:`requirements.yml`
+file - see
+https://github.com/openmicroscopy/prod-playbooks/blob/master/requirements.yml 
+and https://github.com/IDR/deployment/blob/master/ansible/requirements.yml
+for examples of such files.
+
+The release of an Ansible role and its deployment to Galaxy release happens
+by using the
+`Travis integration <https://docs.ansible.com/ansible/latest/reference_appendices/galaxy.html#travis-integrations>`_ on release tag.
```

Comment:
Maybe wrap line. Also not sure about this "on release tag". Is it "on the release tag" or "on tagging the release" maybe?